x-pack/plugins/code/public/sagas/file.ts

```diff
@@ -54,6 +55,8 @@ function* handleFetchRepoTree(action: Action<FetchRepoTreePayload>) {
         // do not request file tree if this tree exists and its children are not empty
         if (!children || children.length === 0) {
           yield call(fetchPath, { uri, revision, path, parents, isDir });
+        } else {
+          yield put(noNeedToFetchRepoTree());
```

Comment:
Is it necessary to apply a new action specifically for setting the loading state to be false? To me, the only thing you need to check if the directory explorer should render the loading icon is to check:

1. if the current path has been loaded?
2. if not, issue the loading request
3. check if file tree is in loading or not?
4. If yes, then check if the current path is in the current fetching path

Assuming that the state only needs to have 2 items:

1. the current fetching path 
2. the file tree 

personally, I think it's a bit redundant to have multiple flags to indicate the loading status of the file tree:

![image](https://user-images.githubusercontent.com/987855/57209004-32c07100-7009-11e9-8e7c-9a894640d441.png)